Doc/c-api/marshal.rst

```diff
@@ -40,20 +40,15 @@ unmarshalling.  Version 2 uses a binary format for floating point numbers.
 
 The following functions allow marshalled values to be read back in.
 
-XXX What about error detection?  It appears that reading past the end of the
-file will always result in a negative numeric value (where that's relevant),
-but it's not clear that negative values won't be handled properly when there's
-no error.  What's the right way to tell? Should only non-negative values be
-written using these routines?
-
 
 .. c:function:: long PyMarshal_ReadLongFromFile(FILE *file)
 
    Return a C :c:type:`long` from the data stream in a :c:type:`FILE\*` opened
    for reading.  Only a 32-bit value can be read in using this function,
    regardless of the native size of :c:type:`long`.
 
-   On error, raise an exception and return ``-1``.
+   On error, sets the appropriate exception (:exc:`EOFError` or
```

Comment:
ValueError can't be raised by the current code. But it can raise MemoryError. In would be nice to make it raising also OSError and KeyboardInterrupt if appropriate.

Reply:
Oops, I definitely missed `MemoryError` (thanks!), but isn't it possible to hit the following branch?

```c
    if (read != n) {
        if (!PyErr_Occurred()) {
            if (read > n)
                PyErr_Format(PyExc_ValueError,
                             "read() returned too much data: "
                             "%zd bytes requested, %zd returned",
                             n, read);
            else
                PyErr_SetString(PyExc_EOFError,
                                "EOF read where not expected");
        }
        return NULL;
    }
```

> It would be nice to make it raising also OSError and KeyboardInterrupt if appropriate.

Is it OK to leave that to another issue? (or is there an already open issue?)

Reply:
> I definitely missed MemoryError

It is missed also in lists of exceptions for `PyMarshal_ReadObjectFromFile()` etc.

> but isn't it possible to hit the following branch?

It looks to me that this branch can be hit only when read from Python file-like object with broken `readinto()`. `PyMarshal_ReadLongFromFile()` reads from the C file, and `fread()` should return `<= n`, unless libc is is broken.

> Is it OK to leave that to another issue? (or is there an already open issue?)

I'm working on this. It will be another issue.

I have doubts that it is worth to specify full lists of exceptions. They almost always will be incomplete.

Reply:
Ok, I will revert to "On error, raise an exception and return -1." in PyMarshal_ReadLongFromFile and PyMarshal_ReadShortFromFile docs.

Do you want me to remove the list of exceptions from the remaining functions?

Reply:
I have no opinion. Providing a full list of exception is unusual, but in this case it may be possible. Look at the history of this documentation, maybe there were special reasons for listing exceptions.

Reply:
Unfortunately, there is not much in the commit logs. The docs for marshal functions were added in https://github.com/berkerpeksag/cpython/commit/0fae49fc7b558836553f2e800f2b9a0bdcf31e45 and they didn't change much since 2001.

I think I'm going to keep EOFError and ValueError and add a general note about MemoryError.

Reply:
Done.